spring-cloud-starter-single-step-batch-job/src/test/java/org/springframework/cloud/task/batch/autoconfigure/FlatFileItemWriterAutoConfigurationTests.java

```diff
@@ -0,0 +1,532 @@
+								BatchAutoConfiguration.class,
+								SingleStepJobAutoConfiguration.class,
+								FlatFileItemWriterAutoConfiguration.class))
+				.withPropertyValues("spring.batch.job.jobName=job",
+						"spring.batch.job.stepName=step1", "spring.batch.job.chunkSize=5",
+						"spring.batch.job.flatfilewriter.name=fooWriter",
+						String.format(
+								"spring.batch.job.flatfilewriter.resource=file://%s",
+								this.outputFile.getAbsolutePath()),
+						"spring.batch.job.flatfilewriter.encoding=UTF-8");
+
+		applicationContextRunner.run((context) -> {
+			JobLauncher jobLauncher = context.getBean(JobLauncher.class);
+
+			Job job = context.getBean(Job.class);
+
+			JobExecution jobExecution = jobLauncher.run(job, new JobParameters());
+
+			JobExplorer jobExplorer = context.getBean(JobExplorer.class);
+
+			while (jobExplorer.getJobExecution(jobExecution.getJobId()).isRunning()) {
+				Thread.sleep(1000);
+			}
+
+			AssertFile.assertLineCount(3, this.outputFile);
+
+			String results = FileCopyUtils.copyToString(new InputStreamReader(
+					new FileSystemResource(this.outputFile).getInputStream()));
+			assertThat(results).isEqualTo("{item=foo}\n{item=bar}\n{item=baz}\n");
+		});
+	}
+
+	@Test
+	public void testHeaderFooterFileGeneration() {
+		ApplicationContextRunner applicationContextRunner = new ApplicationContextRunner()
+				.withUserConfiguration(HeaderFooterConfiguration.class)
+				.withConfiguration(
+						AutoConfigurations.of(PropertyPlaceholderAutoConfiguration.class,
+								BatchAutoConfiguration.class,
+								SingleStepJobAutoConfiguration.class,
+								FlatFileItemWriterAutoConfiguration.class))
+				.withPropertyValues("spring.batch.job.jobName=job",
+						"spring.batch.job.stepName=step1", "spring.batch.job.chunkSize=5",
+						"spring.batch.job.flatfilewriter.name=fooWriter",
+						String.format(
+								"spring.batch.job.flatfilewriter.resource=file://%s",
+								this.outputFile.getAbsolutePath()),
+						"spring.batch.job.flatfilewriter.encoding=UTF-8",
+						"spring.batch.job.flatfilewriter.delimited=true",
+						"spring.batch.job.flatfilewriter.names=item");
+
+		applicationContextRunner.run((context) -> {
+			JobLauncher jobLauncher = context.getBean(JobLauncher.class);
+
+			Job job = context.getBean(Job.class);
+
+			JobExecution jobExecution = jobLauncher.run(job, new JobParameters());
+
+			JobExplorer jobExplorer = context.getBean(JobExplorer.class);
+
+			while (jobExplorer.getJobExecution(jobExecution.getJobId()).isRunning()) {
+				Thread.sleep(1000);
+			}
+
+			AssertFile.assertLineCount(5, this.outputFile);
+
+			String results = FileCopyUtils.copyToString(new InputStreamReader(
+					new FileSystemResource(this.outputFile).getInputStream()));
+			assertThat(results).isEqualTo("header\nfoo\nbar\nbaz\nfooter");
+		});
+	}
+
+	@Configuration
+	@EnableBatchProcessing
+	public static class DelimitedJobConfiguration {
+
+		@Bean
```

Comment:
Probably create a parent class that has the ListItemReader as a Bean.   Addds some reusability.